prov/rxm/src/rxm_cq.c

```diff
@@ -94,6 +94,46 @@ static void rxm_cq_log_comp(uint64_t flags)
 }
 #endif
 
+#define RXM_DEFINE_CNTR_COMP(type)								\
+static inline void rxm_cntr_report_## type ##comp(uint64_t flags, struct rxm_ep *rxm_ep)	\
+{												\
+	struct util_cntr *cntr = NULL;								\
+												\
+	flags &= (FI_SEND | FI_RECV |								\
+		  FI_WRITE | FI_READ |								\
+		  FI_REMOTE_READ | FI_REMOTE_WRITE);						\
+												\
+	switch (flags) {									\
```

Comment:
We shouldn't switch on flags. Flags requires if-else if comparisons.  That said, in most or all cases, the calling site knows which counter to update and should access it directly.  See below.